### PR TITLE
[joi] Update to v13

### PR DESCRIPTION
**Pull request by rokoroku**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  https://github.com/hapijs/joi/issues/1302 v11 Release Note
  https://github.com/hapijs/joi/issues/1326 v12 Release Note
  https://github.com/hapijs/joi/issues/1328 v13 Release Note
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/joi/index.d.ts
Comment (mhegazy) [inline types/joi/index.d.ts]: since this is a major version upgrade, please add a new folder for the old version first. see https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
@rokoroku Please address comments from the code reviewers.

**Comment by RyanCavanaugh** [comment]
@mhegazy - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Pull request merged**
